src/twisted/test/test_defer.py

```diff
@@ -676,6 +676,257 @@ def test_cancelGatherResultsWithAllDeferredsCallback(self):
         self.assertEqual(callbackResult[1], "Callback Result")
 
 
+    def test_extractCurrentResult(self):
+        """
+        L{defer.extractCurrentResult} extracts the results from a Deferred that
+        has already had its callback function called.
+        """
+        result = object()
+
+        d = defer.Deferred()
+        d.callback(result)
+
+        self.assertIs(defer.extractCurrentResult(d), result)
+
+
+    def test_extractCurrentResultRaisesFailureException(self):
+        """
+        L{defer.extractCurrentResult} raises the original exception if the
+        result of the deferred is a failure.
+        """
+        d = defer.Deferred()
+        d.errback(ValueError())
+
+        self.assertRaises(ValueError, defer.extractCurrentResult, d)
+
+
+    def test_extractCurrentResultReturnsFailureWithNoRaises(self):
+        """
+        L{defer.extractCurrentResult} returns the failure instance if
+        raises=False.
+        """
+        d = defer.Deferred()
+        d.errback(ValueError())
+
+        result = defer.extractCurrentResult(d, raises=False)
+
+        self.assertIsInstance(result, failure.Failure)
+        self.assertIsInstance(result.value, ValueError)
+
+
+    def test_extractCurrentResultWithoutResultsNotCaughtByException(self):
+        """
+        L{defer.extractCurrentResult} raises a L{defer.NoCurrentResult}
+        exception which is not caught when catching L{Exception}.
+        """
+        def inner(d):
+            try:
+                defer.extractCurrentResult(d)
+            except Exception:
+                return "error"
+            return "success"
+
+        d = defer.Deferred()
+        self.assertRaises(defer.NoCurrentResult, inner, d)
+
+        # In order to get 100% patch coverage we need to excerise the "error"
```

Comment:
Do we not use `pragma: no cover`? If we do we should just use that for this specific case instead of carrying useless test code IMO.

Reply:
If NoCurrentResult changes to be an Exception then this whole thing goes away!  Yay.  Also there's a typo in "excerise".